dags/analyzer_helper/discourse/transform_raw_info.py

```diff
@@ -0,0 +1,93 @@
+from analyzer_helper.discourse.utils.convert_date_time_formats import DateTimeFormatConverter
+
+class TransformRawInfo:
+
+    def __init__(self):
+        self.converter = DateTimeFormatConverter()
+    
+    def create_data_entry(self, raw_data: list, interaction_type: str = None, interaction_user: int = None):
+        metadata = {
+            "channel_id": raw_data.get("category_id"),
+            "thread_id": raw_data.get("topic_id"),
+            "bot_activity": False,
+        }
+
+        if interaction_type == "reaction":
```

Comment:
for the case of interaction being `reaction` aren't we creating the `receiver` type?

Reply:
@polux0 please answer

Reply:
@amindadgar Apologies for the late response. I think you are right. 

Here is example from the test case: 

    def test_transform_data_with_reactions(self):
            raw_data = [
                {
                    "post_id": 6261,
                    "author_id": 6168,
                    "created_at": "2023-09-11T21:42:43.553Z",
                    "author_name": "Test Author Name2",
                    "reactions": [1, 2],
                    "replied_post_id": None,
                    "replied_post_user_id": None,
                    "topic_id": 6134
                }

            ]
            expected_result = [
                {
                    'author_id': '6168',
                    'date': datetime.datetime(2023, 9, 11, 21, 42, 43, 553000, tzinfo=datetime.timezone.utc),
                    'source_id': '6261',
                    'metadata': {
                        'category_id': None,
                        'topic_id': 6134,
                        'bot_activity': False,
                    },
                    'actions': [
                        {
                            'name': 'message',
                            'type': 'emitter',
                        }
                    ],
                    'interactions': [
                        {
                            'name': 'reaction',
                            'type': 'receiver',
                            'users_engaged_id': ['1', '2'],
                        }
                    ]
                },
                {
                    'actions': [],
                    'author_id': '1',
                    'date': datetime.datetime(2023, 9, 11, 21, 42, 43, 553000, tzinfo=datetime.timezone.utc),
                    'interactions': [
                        {
                            'name': 'reaction',
                            'type': 'emitter',
                            'users_engaged_id': ['6168'],
                        }
                    ],
                    'metadata': {
                        'category_id': None,
                        'topic_id': 6134,
                        'bot_activity': False,
                    },
                    'source_id': '6261',
                },
                {
                    'actions': [],
                    'author_id': '2',
                    'date': datetime.datetime(2023, 9, 11, 21, 42, 43, 553000, tzinfo=datetime.timezone.utc),
                    'interactions': [
                        {
                            'name': 'reaction',
                            'type': 'emitter',
                            'users_engaged_id': ['6168'],
                        }
                    ],
                    'metadata': {
                        'category_id': None,
                        'topic_id': 6134,
                        'bot_activity': False,
                    },
                    'source_id': '6261',
                }
            ]

            result = self.transformer.transform(
                raw_data=raw_data,
            )
            self.assertEqual(result, expected_result)
 
 This looks good to me, but I might be wrong?